core/iwasm/fast-jit/fe/jit_emit_numberic.c

```diff
@@ -472,16 +472,110 @@ jit_compile_op_i64_bitwise(JitCompContext *cc, IntBitwise bitwise_op)
     return false;
 }
 
+static bool
+compile_int_shift(JitCompContext *cc, IntShift shift_op, bool is_i32)
+{
+#if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
+    JitReg ecx_hreg = jit_codegen_get_hreg_by_name("ecx");
+    JitReg rcx_hreg = jit_codegen_get_hreg_by_name("rcx");
+#endif
+    JitReg left, right, mod_right, res;
+
+    POP_INT(right);
+    POP_INT(left);
+
+    if (jit_reg_is_const(right)) {
+        if ((is_i32 && (0 == jit_cc_get_const_I32(cc, right) % 32))
+            || (!is_i32 && (0 == jit_cc_get_const_I64(cc, right) % 64))) {
+            res = left;
+            goto shortcut;
+        }
+    }
+
+    /* right modulo N */
+    if (is_i32) {
+        mod_right = jit_cc_new_reg_I32(cc);
+        GEN_INSN(REM_U, mod_right, right, NEW_CONST(I32, 32));
```

Comment:
When right is const, don't need to GEN_INSN, just `mod_right = NEW_CONST(I32, jit_jit_cc_get_const_I32(cc, right) % 32)`, shall we add the optimization?

And should fix the code guideline check failure.